pkg/chunkenc/pool_test.go

```diff
@@ -50,11 +50,10 @@ func TestPool(t *testing.T) {
 
 	wg.Wait()
 
-	runtime.GC()
-	time.Sleep(20 * time.Millisecond)
-	runtime.GC()
-
-	if !assert.LessOrEqual(t, runtime.NumGoroutine(), 100) {
+	if !assert.Eventually(t, func() bool {
+		runtime.GC()
+		return runtime.NumGoroutine() <= 50
```

Comment:
Curious, why it's 50?

Reply:
It's below 100 and account for other test running in parallel.

Reply:
If there was a leak it would be higher than 200 or more.

Reply:
Hmm. thanks for explaining! Doesn't this also mean, this number will become flaky as we keep adding more and more parallel tests somewhere else?

(not a blocker for the PR)

Reply:
Yeah That's why I hesitate to get it in the first place, but then I thought there was a leak again. So I wrote a test. Normally go should run that many test in parallel though, it's also only possible for the current package.